File: app/src/main/java/com/example/applligent/nagoriengineering/view/company/DetailsActivity.java
package com.example.applligent.nagoriengineering.view.company;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.view.View;

import com.example.applligent.nagoriengineering.MyNagoriApplication;
import com.example.applligent.nagoriengineering.R;
import com.example.applligent.nagoriengineering.dao.ItemDao;
import com.example.applligent.nagoriengineering.databinding.ActivityDetailsBinding;
import com.example.applligent.nagoriengineering.model.Item;

public class DetailsActivity extends AppCompatActivity {
    DrawerLayout drawerLayout;
    long itemId;
    ItemDao itemDao;
    ActivityDetailsBinding binding;
    Item item;

    public int partsId;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_details);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle("Detail");
            actionBar.setHomeAsUpIndicator(R.drawable.ic_arrow_back);
            actionBar.setDisplayShowTitleEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
        itemDao = MyNagoriApplication.getDatabase().itemDao();
        itemId = getIntent().getLongExtra("item_id", -1);
        showDetails();
        binding.parts.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                item = itemDao.get(itemId);
                Intent intent = new Intent(getApplicationContext(), PartsInfoActivity.class);
                intent.putExtra("tel_part_number", item.getTelPartNumber());
                startActivity(intent);

            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) finish();
        return super.onOptionsItemSelected(item);
    }

    public void showDetails() {
        item = itemDao.get(itemId);
        binding.telPartLabel.setText(item.getTelPartNumber());
        binding.oemPartLabel.setText(item.getOem());
        binding.engineLabel.setText(item.getEngine());
        binding.applicationLabel.setText(item.getApplication());
        binding.mrpLabel.setText(Float.toString(item.getMrp()));
        binding.telPartLabel.setText(item.getTelPartNumber());
        binding.alphaLabel.setText(Integer.toString(item.getOrientationAlpha()));
        binding.betaLabel.setText(Integer.toString(item.getOrientationBeta()));
        binding.strokingLabel.setText(item.getStrPre());
        binding.settingPrLabel.setText(item.getSettingPre());
        binding.liftLabel.setText(item.getLift());
        binding.remarksLabel.setText(item.getReman());
    }

    public void showServiceInfo(){

        partsId = getIntent().getIntExtra("parts_info", -1);
        item=itemDao.get(partsId);

    }
}

File: app/src/main/java/com/example/applligent/nagoriengineering/view/company/DetailsActivity.kt
package com.example.applligent.nagoriengineering.view.company

import android.content.Intent

import android.os.Bundle
import android.support.v4.widget.DrawerLayout
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import android.view.View

import com.example.applligent.nagoriengineering.MyNagoriApplication
import com.example.applligent.nagoriengineering.R
import com.example.applligent.nagoriengineering.dao.ItemDao
import com.example.applligent.nagoriengineering.model.Item
import kotlinx.android.synthetic.main.activity_details.*

class DetailsActivity : AppCompatActivity() {
    lateinit var drawerLayout: DrawerLayout
     var itemId: Long =0
    lateinit var itemDao: ItemDao

    lateinit var item: Item

    var partsId: Int = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
       setContentView(R.layout.activity_details)
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.title = "Detail"
            actionBar.setHomeAsUpIndicator(R.drawable.ic_arrow_back)
            actionBar.setDisplayShowTitleEnabled(true)
            actionBar.setDisplayHomeAsUpEnabled(true)
        }
        itemDao = MyNagoriApplication.getDatabase(applicationContext).itemDao()
        itemId = intent.getLongExtra("item_id", -1)
        showDetails()
        parts.setOnClickListener {
            item = itemDao.get(itemId)
            val intent = Intent(applicationContext, PartsInfoActivity::class.java)
            intent.putExtra("tel_part_number", item.telPartNumber)
            startActivity(intent)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) finish()
        return super.onOptionsItemSelected(item)
    }

    fun showDetails() {
        item = itemDao.get(itemId)
       telPartLabel.text = item.telPartNumber
       oemPartLabel.text = item.oem
       engineLabel.text = item.engine
       applicationLabel.text = item.application
       mrpLabel.text = java.lang.Float.toString(item.mrp)
       telPartLabel.text = item.telPartNumber
       alphaLabel.text = Integer.toString(item.orientationAlpha)
       betaLabel.text = Integer.toString(item.orientationBeta)
       strokingLabel.text = item.strPre
       settingPrLabel.text = item.settingPre
       liftLabel.text = item.lift
       remarksLabel.text = item.reman
    }

    fun showServiceInfo() {

        partsId = intent.getIntExtra("parts_info", -1)
        item = itemDao.get(partsId.toLong())

    }
}




